cpp/src/column/column.cu

```diff
@@ -275,6 +278,93 @@ struct create_column_from_view_vector {
 
 };
 
+struct concatenate_using_partition_map {
+  std::vector<column_view> const& views;
+  cudaStream_t stream;
+  rmm::mr::device_memory_resource* mr;
+
+  template <typename T,
+      std::enable_if_t<is_fixed_width<T>()>* = nullptr>
+  std::unique_ptr<column> operator()() {
+
+    // Compute the partition offsets
+    thrust::host_vector<size_type> offsets(views.size() + 1);
+    thrust::transform_inclusive_scan(thrust::host,
+      views.begin(), views.end(), offsets.begin() + 1,
+      [](column_view const& col) {
+        return col.size();
+      },
+      thrust::plus<size_type>{});
+    thrust::device_vector<size_type> d_offsets(offsets);
+    auto const output_size = offsets.back();
+
+    // Compute partition map
+    // NOTE instead of scattering `i` and doing a "max scan", scatter 1s
```

Comment:
@jrhemstad You had suggested doing a max scan, but I wasn't able to find an equivalent of this in thrust. Do you know another way of doing this? I _think_ doing an inclusive scan instead should be just as efficient, but I'm curious.

Reply:
https://thrust.github.io/doc/group__prefixsums_ga7109170b96a48fab736e52b75f423464.html

You can pass `thrust::maximum` as the binary op.

I think what you did here should be just as efficient though, so there's not really a need to change it.

Reply:
Oh duh, I wasn't considering that you can pass your own binary op here. Ok, seems fine either way.